### PR TITLE
chore: ts strict fixes for ui theme

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0618 — APPROVED** [review]

File: packages/ui/src/theme/utils.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function isDesignToken(value: unknown): value is WebDesignToken {
   return isObject(value) && has(value, 'value');
 }
 
-export function isShadowToken(value: unknown): value is ShadowValue {
+export function isShadowToken(value: unknown): value is ShadowValue & object {
```
Comment (calebpollman) [inline packages/ui/src/theme/utils.ts:68]: Fairly certain that the definition of `ShadowValue` can resolve to either an `object` of shadow props or `string`. Not fully related to this PR but maybe we should rename this to something more explicit

File: packages/ui/src/theme/types/style-dictionary.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+declare module 'style-dictionary/lib/utils/references/usesReference.js' {
+  // based on the typing specified by style-dictionary:
+  // https://github.com/amzn/style-dictionary/blob/main/lib/utils/references/usesReference.js
+  export default function usesReference(value: string): boolean;
```
Comment (reesscot) [inline packages/ui/src/theme/types/style-dictionary.d.ts:4]: Is there potential for drift on these types? I'm assuming these types cannot be exported from style-dictionary, is that correct?

**Review by reesscot — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]

File: packages/ui/src/theme/utils.ts (hunk 1 above)
Comment (dbanksdesign) [reply packages/ui/src/theme/utils.ts:68]: good point, `isShadowObject` it is

File: packages/ui/src/theme/types/style-dictionary.d.ts (hunk 2 above)
Comment (dbanksdesign) [reply packages/ui/src/theme/types/style-dictionary.d.ts:4]: That is correct, something if I ever get time to do I would like to rewrite the whole thing in TS and expose the correct types with better exports so we don't have to import specific files. There are a lot of projects that do this so I promise not to break any of these paths :)

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]

**Pull request merged**